tests/test_transformer.py

```diff
@@ -2090,3 +2141,86 @@ def add(a, b):
         def add(a: rt_T1, b: rt_T1) -> rt_T1:
             return a + b
     """)
+
+
+def test_overload_preserve():
+    code = cst.parse_module(textwrap.dedent("""\
+        from typing import overload
+
+        @overload
+        def foo(bar: int) -> str:
+            ...
+        @overload
+        def foo(bar: str) -> int:
+            ...
+        def foo(bar):
+            if isinstance(bar, int):
+                return "hello"
+            elif isinstance(bar, str):
+                return 2
+            elif isinstance(bar, bool):
+                return not bar
+    """))
+    T1 = TypeInfo.from_type(str)
+    T2 = TypeInfo.from_type(int)
+    f = get_funcid('foo.py', code, 'foo')
+    t = UnifiedTransformer(
+            filename='foo.py',
+            type_annotations = {
+                f: FuncAnnotation(
+                    [
+                        (ArgumentName("bar"), T1),
+                    ],
+                    T2,
+                ),},
+            override_annotations=False,
+            only_update_annotations=False,
+            inline_generics=False,
+            module_name='foo',
+            module_names=['foo'],
+        )
+
+    code = t.transform_code(code)
+    assert "@overload" in "\n".join(get_function_all(code, "foo"))
```

Comment:
I think it would be better to save the original code (out of the `dedent`) and assert that the transformed code is the same...